### PR TITLE
Add table of contents to README

### LINKED ISSUE #335: Add Table of Contents section to the README
As a rack-attack user
I want the README to have a Table of Contents with hyperlinks to each of the sections
So that I can jump directly to the part of the README I am interested in

### PR DESCRIPTION
**Pull request by leoarnold**
Closes #335

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,9 +10,42 @@ See the [Backing & Hacking blog post](https://www.kickstarter.com/backing-and-ha
...
 
+## Table of contents
+
+- [Getting started](#getting-started)
+  - [Installing](#installing)
+  - [Plugging into the application](#plugging-into-the-application)
+- [Usage](#usage)
+  - [Safelisting](#safelisting)
+    - [`safelist_ip(ip_address_string)`](#safelistipipaddressstring)
+    - [`safelist_ip(ip_subnet_string)`](#safelistipipsubnetstring)
```
Comment (grzuy) [inline README.md]: These two links above not working for me

File: README.md [hunk 2]
```diff
@@ -10,9 +10,42 @@ See the [Backing & Hacking blog post](https://www.kickstarter.com/backing-and-ha
...
+    - [`safelist(name, &block)`](#safelistname-block)
+  - [Blocking](#blocking)
+    - [`blocklist_ip(ip_address_string)`](#blocklistipipaddressstring)
+    - [`blocklist_ip(ip_subnet_string)`](#blocklistipipsubnetstring)
```
Comment (grzuy) [inline README.md]: These two above also not working for me

File: README.md [hunk 3]
```diff
@@ -10,9 +10,42 @@ See the [Backing & Hacking blog post](https://www.kickstarter.com/backing-and-ha
...
+    - [`blocklist(name, &block)`](#blocklistname-block)
+    - [Fail2Ban](#fail2ban)
+    - [Allow2Ban](#allow2ban)
+  - [Throttling](#throttling)
+    - [`throttle(name, options, &block)`](#throttlename-options-block)
+  - [Tracks](#tracks)
+  - [Cache store configuration](#cache-store-configuration)
+- [Customizing responses](#customizing-responses)
+  - [RateLimit headers for well-behaved clients](#ratelimit-headers-for-well-behaved-clients)
+- [Logging & Instrumentation](#logging-instrumentation)
```
Comment (grzuy) [inline README.md]: Not working for me

**Comment by leoarnold** [comment]
Fixed - thanks for checking!

**Comment by grzuy** [comment]
Thank you! :rocket:

**Pull request merged**